tests/pytorch/test_cpu_offloading.py

```diff
@@ -2,84 +2,143 @@
 #
 # See LICENSE for license information.
 
+import os
+from contextlib import nullcontext
 import pytest
 import torch
-from contextlib import nullcontext
 
 import transformer_engine.pytorch as te
+from transformer_engine.common import recipe
 from transformer_engine.pytorch.fp8 import FP8GlobalStateManager
 
-# Check if FP8 supported
+# Check if FP8 is supported
 fp8_available, reason_for_no_fp8 = FP8GlobalStateManager.is_fp8_available()
+mxfp8_available, reason_for_no_mxfp8 = FP8GlobalStateManager.is_mxfp8_available()
+
+fp8_recipes = [
+    None,  # non-fp8
+    # recipe.MXFP8BlockScaling(), - scale inverse tensors offloading doest not work yet
+    recipe.Float8CurrentScaling(),
+    recipe.DelayedScaling(),
+]
 
 SIZE = 512
+NUM_HEADS = 8
+NUM_LAYERS = 5
+EPSILON = 0.1
```

Comment:
For an FP8 linear layer, the cached input is 32 MiB and the weight cache is 0.5 MiB. A 0.1 MiB tolerance seems sensible, although we may want to revisit if we ever reduce the data dimensions. A relative tolerance would be more generalizable, but I don't think it's necessary.